### ISSUE TITLE
Feature Request: Improve the friction of confirmDelete pop-up

### ISSUE THREAD

**Issue by byronhulcher** [clay/clay-kiln#1177]
confirmDelete is good but we're still finding that people are removing things they shouldn't be. A simple solution could be adding more requirements/steps/details to the confirm/delete.  Some ideas:

- Asking for a required "Reason why you are deleting this" that we can send to google analytics for our report
- Multi-step prompt ("Are you REALLY REALLY sure?!")
- Make it redder/angrier
- Make them solve a circa-2006 escape the room flash puzzle game

**Comment by nelsonpecora** [comment]
Per discussion, we should _replace_ the "type the component's name" (github-style) friction with a simple "reason for removal" textarea.

**Comment by nelsonpecora** [comment]
@reubenson you expressed interest in working on this?

**Comment by nelsonpecora** [comment]
@pfeytser what language did you want in this?

**Comment by reubenson** [comment]
@pfeytser also, i'll need to assign a custom dimension to store the user input data. `dimension2` sound good to you?

**Comment by pfeytser** [comment]
@nelsonpecora @reubenson 
                                              
How about: 

This component was marked as critical and is not frequently deleted. Please specify a reason for removal.

**Comment by pfeytser** [comment]
Use cd-2 (user input) for tracking. 

![Screen Shot 2018-04-30 at 10.36.04 AM.png](https://images.zenhubusercontent.com/5a05f7e38a75884b9086f30a/76a5d5d5-e683-4267-bf04-fe7065384627)

**Comment by reubenson** [comment]
@pfeytser `This component was marked as critical and is not frequently deleted. Please specify a reason for removal.` works for me. The existing `confirmRemoval` modal also states `Are you sure you want to remove this [component]?` - I think we should also keep this, and add the additional statement?

**Comment by pfeytser** [comment]
@reubenson I would put `Are you sure you want to remove this [component]?` at the beginning rather than the end.

**Comment by reubenson** [comment]
@pfeytser yup, agree with your comment above.  Here's a QA link - http://confirm-reason.nymag.sites.aws.nymetro.com/strategist/_pages/cjh0w9d2z006bery6i71cezwa.html?edit=true
